src/components/common/Calculator.js

```diff
@@ -108,39 +141,46 @@ class Calculator extends Component {
   }
 
   updateAngleUnit = (angleUnit) => {
+    const isRadian = angleUnit === ANGLE_UNITS.RAD;
+
     parser.set('tan', (content) => {
-      // set corner cases because of rounding error
-      const angle = (content / 90.0) % 4;
-      if (angleUnit === ANGLE_UNITS.DEG) {
-        if (angle === 1 || angle === 3) {
-          return 'Infinity';
-        }
-        if (angle === -1 || angle === -3) {
-          return '-Infinity';
-        }
-      }
-      return math.tan(math.unit(content, angleUnit));
+      const value = getSpecialCase(
+        content,
+        TRIGONOMETRY_FUNCTIONS.TAN,
+        isRadian,
+      );
+      return value !== false ? value : math.tan(math.unit(content, angleUnit));
```

Comment:
Not sure what this ternary operation is doing. Same for the next two cases.

Reply:
I first evaluate whether a value is returned from the special case handler. If yes I return the special value. If not I compute it "manually".